### PR TITLE
Upgrade Akka HTTP, fix and turn on integration tests

### PR DESCRIPTION
**Pull request by richdougherty**
Even though Akka HTTP is on a new version I've kept the other Akka libraries on version 2.3.9 due to #4381.

Commits are split into several logical parts so don't need further squashing. More info in the commit messages.

### CODE REVIEW

File: framework/src/play-integration-test/src/test/scala/play/it/http/ScalaResultsHandlingSpec.scala [hunk 1]
```diff
@@ -107,7 +107,7 @@ trait ScalaResultsHandlingSpec extends PlaySpecification with WsTestClient with
           BasicRequest("GET", "/", "HTTP/1.0", Map("Connection" -> "keep-alive"), "")
         )(0)
         response.status must_== 200
-        response.headers.get(CONNECTION) must beNone
+        response.headers.get(CONNECTION).map(_.toLowerCase) must beOneOf(None, Some("close"))
```
Comment (marcospereira) [inline framework/src/play-integration-test/src/test/scala/play/it/http/ScalaResultsHandlingSpec.scala]: Will `"close"` be a standard message for close the connection?

File: framework/src/play-integration-test/src/test/scala/play/it/http/ScalaResultsHandlingSpec.scala [hunk 2]
```diff
@@ -346,7 +346,7 @@ trait ScalaResultsHandlingSpec extends PlaySpecification with WsTestClient with
           BasicRequest("GET", "/", "HTTP/1.1", Map(), "")
         ).apply(0)
         response.status must_== Status.INTERNAL_SERVER_ERROR
-        (response.headers - (CONTENT_LENGTH)) must be(Map.empty)
-      }.pendingUntilAkkaHttpFixed // https://github.com/akka/akka/issues/16988
+        (response.headers -- Seq(CONTENT_LENGTH, SERVER, DATE, CONNECTION)) must be(Map.empty)
```
Comment (marcospereira) [inline framework/src/play-integration-test/src/test/scala/play/it/http/ScalaResultsHandlingSpec.scala]: What is the expectation here?

1. Have **only** these headers
2. Have **all** of these headers

?

File: framework/src/play-integration-test/src/test/scala/play/it/http/ScalaResultsHandlingSpec.scala (hunk 1 above)
Comment (richdougherty) [inline framework/src/play-integration-test/src/test/scala/play/it/http/ScalaResultsHandlingSpec.scala]: `close` is a valid value for the `Connection` header. I believe `close` is the default in HTTP 1.0, so it's assumed to be `close` if the header isn't set. We're not setting the header with Netty, but Akka HTTP has decided to explicitly send the value. Either is OK so I'm using `beOneOf` to allow both.

File: framework/src/play-integration-test/src/test/scala/play/it/http/ScalaResultsHandlingSpec.scala (hunk 2 above)
Comment (richdougherty) [inline framework/src/play-integration-test/src/test/scala/play/it/http/ScalaResultsHandlingSpec.scala]: Any or all of these headers are optional. No other headers are expected or allowed.

The main thing is the status check.

File: framework/src/play-integration-test/src/test/scala/play/it/http/ScalaResultsHandlingSpec.scala (hunk 1 above)
Comment (marcospereira) [inline framework/src/play-integration-test/src/test/scala/play/it/http/ScalaResultsHandlingSpec.scala]: Got it. This spec is executed for both Netty and Akka HTTP. :+1:

Comment (richdougherty) [inline framework/src/play-integration-test/src/test/scala/play/it/http/ScalaResultsHandlingSpec.scala]: Yes, that's right!

**Comment by richdougherty** [comment]
Rebased because of merge conflicts.

**Comment by richdougherty** [comment]
Hopefully rebased properly this time!

**Comment by richdougherty** [comment]
This is now ready to be merged.

**Pull request merged**